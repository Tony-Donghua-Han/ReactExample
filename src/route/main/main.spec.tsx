import React from "react";
import { shallow } from "enzyme";
import Main from "./main.component";

import { AuthStatus } from "../../redux/Auth/auth.reducer";

describe("Main", (): void => {
    test("noUser state should render without crushing", (): void => {
        const wrapper = shallow(<Main status={AuthStatus.noUser} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test("authenticatingUser state should render without crushing", (): void => {
        const wrapper = shallow(<Main status={AuthStatus.authenticatingUser} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test("userAuthenticated state should render without crushing", (): void => {
        const wrapper = shallow(<Main status={AuthStatus.userAuthenticated} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
