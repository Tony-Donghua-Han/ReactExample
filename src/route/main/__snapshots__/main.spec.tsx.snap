// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main authenticatingUser state should render without crushing 1`] = `
"<div className=\\"main\\" id=\\"main\\">
  <LoginBox />
</div>"
`;

exports[`Main noUser state should render without crushing 1`] = `
"<div className=\\"main\\" id=\\"main\\">
  <LoginBox />
</div>"
`;

exports[`Main userAuthenticated state should render without crushing 1`] = `
"<div className=\\"main\\" id=\\"main\\">
  Customer name is 
</div>"
`;
